### PR TITLE
EnginePrimeExportJob: Use mixxx::audio::FramePos internally

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: src/library/export/engineprimeexportjob.cpp [hunk 1]
```diff
@@ -177,10 +167,10 @@ void exportMetadata(djinterop::database* pDatabase,
     snapshot.average_loudness = pTrack->getReplayGain().getRatio();
 
     // Set main cue-point.
-    double cuePlayPos = pTrack->getCuePoint().getPosition();
-    double cueSampleOffset = playPosToSampleOffset(cuePlayPos);
-    snapshot.default_main_cue = cueSampleOffset;
-    snapshot.adjusted_main_cue = cueSampleOffset;
+    mixxx::audio::FramePos cuePlayPos = pTrack->getMainCuePosition();
+    // FIXME: What if the cuePlayPos is invalid?
```
Comment (uklotzde) [inline src/library/export/engineprimeexportjob.cpp:171]: @mr-smidge Please verify and handle these cases after we merged this PR.

File: src/library/export/engineprimeexportjob.cpp [hunk 2]
```diff
@@ -204,14 +193,13 @@ void exportMetadata(djinterop::database* pDatabase,
         // number of beats from there until the end of the track in order to
         // correctly assign an index for the last beat.
         const auto lastBeatPlayPos =
-                beats->findPrevBeat(mixxx::audio::FramePos::fromEngineSamplePos(
-                        sampleOffsetToPlayPos(sampleCount)));
+                beats->findPrevBeat(mixxx::audio::FramePos(frameCount));
```
Comment (uklotzde) [inline src/library/export/engineprimeexportjob.cpp:196]: `mixxx::audio::kStartFramePose + frameCount`? Doesn't matter but would be the correct calculation.

**Review by uklotzde — APPROVED** [review]
LGTM

File: src/library/export/engineprimeexportjob.cpp (hunk 1 above)
Comment (mr-smidge) [reply src/library/export/engineprimeexportjob.cpp:171]: Under what circumstances can the main cue position be invalid?  If not set?

I suppose the simplest solution is just to set to zero/start-of-track in the export.

**Pull request merged**